### ISSUE TITLE
[IDEA] Short ascii alias to tiddlers with long, unicode titles

### ISSUE THREAD

**Issue by shardator** [TiddlyWiki/TiddlyWiki5#6342]
**Is your feature request related to a problem? Please describe.**

I have a small wiki in Hungarian, and there are many accented letters in the language. I also want to have titles decorated with emojis, add commas, etc. But I'd also like to have a way to simply link them, without having to copy the whole title.

Sometimes, when I change the title to this new format, the references work, but sometimes they get broken. For example the link

```
#ArvizturoTukorfurogep
```

opens a new empty tiddler with the same title, but I'd like to change the title to

```
Árvíztűrő™ Tükörfúrógép®
```

sometimes this works, but more often it does not. **And when it does not, it is totally silent, the link gets lost, the new page gets orpaned, unreachable.** Clicking the link opens a new empty tiddler again.

**Describe the solution you'd like**

I'd simply like to have a way to specify **both** a visible title and a **wikiword** for the same tiddler.

**Describe alternatives you've considered**

None.

**Additional context**

None.

**Comment by shardator** [comment]
OK, there is a solution:

> As for differently named links opening the same tiddler, Mario has made a uni-link plugin that does this using I believe aliases. I have not used it, but many others do.

**Comment by shardator** [comment]
The only problem with UniLink that it is a bit tedious (one has to carefully set the aliases, they are not filled automatically clicking on the link for a non-existing page), **and does not work with markdown!**. Is there a way around the latter?

**Comment by pmario** [comment]
>The only problem with UniLink that it is a bit tedious (one has to carefully set the aliases, they are not filled automatically clicking on the link for a non-existing page),

There has been a discussion lately, that the "Missing tiddler "xxxx" – click to create" template doesn't allow us to create markdown tiddlers. The default action will always create a standard TW tiddler. see: **tm-edit-tiddler should take (and use) a bag of fields like tm-new-tiddler** #6001 

The idea there was, that the markdown plugin could add a template, that would be show as an additional text eg: "Missing tiddler "xxxx" – click to create a markdown tiddler" 

The "edit tiddler" toolbar button would also need a possibility to create an MD or TW tiddler .. 

It may be possible, that uni-link also brings a "template" that shows "Create New Alias Tiddler" with a prefilled **aliases** field. ... 

I think this would be valid extension for the plugin.

**Comment by pmario** [comment]
I did add a new issue to the uni-link repo: https://github.com/wikilabs/plugins/issues/108

**Comment by pmario** [comment]
>and does not work with markdown!. Is there a way around the latter?

I'm not sure about this one. ... Did you mean: an `[[alias-name|?]]` link in a markdown tiddler link to the alias? So similar to a TW "pretty link". 

Similar to this discussion? ** [BUG] The new markdown plugin doesn't support Wikitext links.** #4610

**Comment by pmario** [comment]
If you add `unilink` to the following config tiddler: https://tiddlywiki.com/plugins/tiddlywiki/markdown/#%24%3A%2Fconfig%2Fmarkdown%2FrenderWikiTextPragma

It's important that you **reload** the page after you added the setting. 

The an alias-link like [[test|?]] should also work in a markdown tiddler. I did test it a minute ago.

**Comment by shardator** [comment]
@pmario Unfortunately it does not seem to be working. Do I do something wrong? Tried with `|?` and `|?t` suffix too.

![Output](https://user-images.githubusercontent.com/15422254/146373694-455a9dd8-f9d0-4209-a363-e8bfafb6b2b0.png)

Sources:

![Sources](https://user-images.githubusercontent.com/15422254/146373818-af14e206-91af-4f38-a1cf-d7a7548e7994.png)

**Comment by pmario** [comment]
Did you **reload** the wiki after adding `unilink` to the config tiddler?

**Comment by shardator** [comment]
Yes. Saved the whole thing into a file and reloaded. https://github.com/shardator/tw-test/raw/main/ttt.html

**Comment by pmario** [comment]
You did _not_ add the `unilink` parameter to https://tiddlywiki.com/plugins/tiddlywiki/markdown/#%24%3A%2Fconfig%2Fmarkdown%2FrenderWikiTextPragma

It needs to look like this: 

![grafik](https://user-images.githubusercontent.com/374655/146380956-5b79bb9e-0763-4bc4-ac72-a15e91a229c4.png)

**Comment by shardator** [comment]
Woks!

![Magic](https://c.tenor.com/wn2_Qq6flogAAAAS/magical-magic.gif)

Can I override this without editing the shadow tiddly?

**Comment by pmario** [comment]
>Can I override this without editing the shadow tiddly?

No -- it's a config tiddler, which is designed to be overwritten.

**Comment by shardator** [comment]
What will happen if I upgrade the plugin to a newer version? Is not the point of shadow tiddlers to be overridable without clashes by non-shadow ones?

**Comment by pmario** [comment]
With a newer version of the plugin you still want the alias-links to work. right? 
So your modified configuration will win over the shadow tiddler of the new plugin. ... That's usually what is expected.

**Comment by pmario** [comment]
Issue was resolved. So close
